app/models/user.rb

```diff
@@ -210,7 +210,7 @@ def valid_confirmation_token?
 
   def generate_confirmation_token(reset_unconfirmed_email: true)
     self.unconfirmed_email = nil if reset_unconfirmed_email
-    self.confirmation_token = Clearance::Token.new
+    self.confirmation_token = SecureRandom.hex(24).encode("UTF-8")
```

Comment:
why is the encode necessary here? the hex should always be valid ascii

Reply:
I was just cargo-culting. I have no idea why this would be an issue but it's what the code used to do. 🤷

Reply:
The `encode` [is for something sqlite related issue](https://github.com/thoughtbot/clearance/commit/e6b5928be50853b7cfee0b182c37e0680807fad6) so i'm confident we can leave it out.

Reply:
For what it's worth, it's all over (after removing the one here).

```
app/avo/actions/reset_user_2fa.rb:      user.password = SecureRandom.hex(20).encode("UTF-8")
app/models/user.rb:    who = who.encode(Encoding::UTF_8)
app/models/user.rb:      password: SecureRandom.hex(20).encode("UTF-8")
app/models/ownership.rb:    self.token = SecureRandom.hex(20).encode("UTF-8")
```